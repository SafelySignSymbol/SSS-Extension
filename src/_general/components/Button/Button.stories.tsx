import React from 'react'
import { Story, Meta } from '@storybook/react'

import Button, { Props } from './index'

export default {
  title: 'Elements/Button',
  component: Button,
} as Meta

const Template: Story<Props> = (args) => <Button {...args} />

export const Default = Template.bind({})
Default.args = {
  text: 'BUTTON',
  onClick: () => console.log('clicked'),
}
