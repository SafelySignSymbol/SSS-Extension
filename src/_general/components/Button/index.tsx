import React from 'react'
import styled from '@emotion/styled'

import { Button } from '@mui/material'
import Color from '../../utils/Color'

export interface Props {
  text: string
  onClick: () => void
}

const Component: React.VFC<Props> = ({ text, onClick }) => {
  return (
    <SButton variant="outlined" onClick={onClick}>
      {text}
    </SButton>
  )
}

export default Component

const SButton = styled(Button)({
  color: Color.blue,
  ':hover': {
    background: Color.blue,
    color: Color.base_white,
  },
})
