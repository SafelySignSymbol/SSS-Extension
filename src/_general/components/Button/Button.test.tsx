import { fireEvent, render, screen } from '@testing-library/react'
import Button from '.'

const expected_label = 'EXPECTED'
const unexpected_label = 'UN EXPECTED'

describe('/components/Button', () => {
  const onClickMock = jest.fn()
  const Target: React.FC<{ text: string }> = ({ text }) => (
    <Button text={text} onClick={onClickMock} />
  )

  describe('適切にレンダーできる', () => {
    const { container } = render(<Target text="TEST" />)
    expect(container).toMatchSnapshot()
  })

  test('ボタンのラベルテキストの一致する場合、真になる', () => {
    render(<Target text={expected_label} />)
    expect(screen.queryByText(expected_label)).toBeTruthy()
  })
  test('ボタンのラベルテキストが一致しない場合、偽になる', () => {
    render(<Target text={expected_label} />)
    expect(screen.queryByText(unexpected_label)).toBeFalsy()
  })
  test('クリック時にハンドラが呼ばれる', () => {
    render(<Target text={expected_label} />)
    fireEvent.click(screen.getByText(expected_label))
    expect(onClickMock).toBeCalled()
  })
})
