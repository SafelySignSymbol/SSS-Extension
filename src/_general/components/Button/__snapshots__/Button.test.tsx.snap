// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1qon96r-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    TEST
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
